protostar/utils/compiler/pass_managers.py

```diff
@@ -38,19 +71,27 @@ def build(config: "CompilerConfig") -> PassManager:
 class TestCollectorPassManagerFactory(StarknetPassManagerFactory):
     @staticmethod
     def build(config: "CompilerConfig") -> PassManager:
-        pass_manager = StarknetPassManagerFactory.build(config)
-        crucial_stages: List[Tuple[str, Stage]] = [
-            stage_pair
-            for stage_pair in pass_manager.stages
-            if stage_pair[0]
-            in [
-                "module_collector",
-                "unique_label_creator",
-                "identifier_collector",
-            ]
-        ]
-        pass_manager.stages = crucial_stages
-        return pass_manager
+        read_module = get_module_reader(cairo_path=config.include_paths).read
+
+        manager = PassManager()
+        manager.add_stage(
+            "module_collector",
+            ModuleCollector(
+                read_module=read_module,
+                additional_modules=[],
+            ),
+        )
+        collector_fac: Callable[
+            [PassManagerContext], Visitor
+        ] = lambda _: TestCollectorPreprocessor()
```

Comment:
by adding a `context` argument to `TestCollectorPreprocessor.__init__` you could omit this lambda trickery